src/irace/__init__.py

```diff
@@ -107,12 +109,20 @@ class irace:
 
     def __init__(self, scenario, parameters_table, target_runner):
         self.scenario = scenario
+        self.instances = scenario.get('instances', None)
+        self.context = {}
         if 'instances' in scenario:
-            self.scenario['instances'] = np.asarray(scenario['instances'])
+            self.context.update({
+                'py_instances': self.scenario['instances'],
+            })
+            self.scenario['instances'] = StrVector(list(map(lambda x: json.dumps(x, skipkeys=True, default=self.scenario.get('instanceObjectSerializer', lambda x: '<not serializable>')), self.scenario['instances'])))
```

Comment:
Does this mean that the serializer is applied to strings and integers? Isn't that a waste?

Also, could you document with a comment what this line is doing? It seems to replace the instance with the string '<not serializable>' unless the user provides an 'instanceObjectSerializer'. Is that the case? If so, how can the user serialize their instances? It is unclear to me how all this works and will be used in practice.

A couple of examples using for example:

* Functions that need to be optimized by https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.minimize.html#scipy.optimize.minimize
* Matrices that are to be optimized by https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.quadratic_assignment.html#scipy.optimize.quadratic_assignment